src/js/providers/tracks-mixin.js

```diff
@@ -551,10 +551,7 @@ define(['utils/underscore',
             // already have one with the same label
             track = _.findWhere(tracks, { label: label });
 
-            if (track) {
-                track.kind = itemTrack.kind;
-                track.language = itemTrack.language || '';
-            } else {
```

Comment:
My guess is that functionality is required for playlists where the video tag and it's tracks are reused.

💉  🙉  `removeTextTrack` we need you!